k8s-kvm/provision_k8s_host_kvm.sh

```diff
@@ -0,0 +1,145 @@
+  echo "Fetching base image:"
+  curl -L https://cf-opensusefs2.s3.amazonaws.com/vagrant/${K8S_VM_IMAGE_NAME}.box | tar --wildcards box.img -xz
+  mv box.img $K8S_VM_IMAGE_NAME.qcow2
+fi
+
+# Fetch VM access key if it doesn't exist
+if [[ ! -f vm-key ]]; then
+  curl -sL -o "vm-key" https://raw.githubusercontent.com/mitchellh/vagrant/v1.9.6/keys/vagrant
+  chmod 400 vm-key
+fi
+K8S_VM_KEY=${K8S_VM_KEY:-$PWD/vm-key} 
+
+if [[ $K8S_VM_MEM_GIB -gt $(( $(free -g | grep [0-9] | head -1 | awk '{print $2}') - 2 )) ]]; then
+  echo K8S_VM_MEM_GIB value $K8S_VM_MEM_GIB exceeds system resources
+  exit 1
+fi
+
+unset RUNNING_VM_IMAGE
+if virsh domstate $K8S_VM_NAME 2>/dev/null; then
+  echo "Deleting existing domain $K8S_VM_NAME"
+  RUNNING_VM_IMAGE=$(virsh dumpxml $K8S_VM_NAME | grep 'source file' | grep -o "'.*'" | tr -d "'")
+  if [[ -z $RUNNING_VM_IMAGE ]]; then
+    echo "Disk for domain $K8S_VM_NAME could not be determined. Please delete this domain manually"
+    exit 1
+  fi
+  virsh destroy "$K8S_VM_NAME"
+  virsh undefine "$K8S_VM_NAME"
+  rm -f "$RUNNING_VM_IMAGE"
+fi
+
+cp "$K8S_VM_IMAGE_NAME.qcow2" "$K8S_VM_IMAGE_NAME-$K8S_VM_NAME.qcow2"
+# TODO: Test if virbr0 is present
+virt-install \
+  --name "$K8S_VM_NAME" \
+  --memory $(( K8S_VM_MEM_GIB * 1024 )) \
+  --vcpus 1 \
+  --disk "$K8S_VM_IMAGE_NAME-$K8S_VM_NAME.qcow2" \
+  --import \
+  --network network=default \
+  --network bridge=br0 \
+  --os-variant opensuse42.2 \
+  --wait 0 \
+  --hvm
+
+echo "Sleeping 30 seconds to allow VM to start and obtain IP"
+sleep 30 
+unset K8S_VM_DEFAULT_IP
+# TODO: Clean up variable assignment (timeouts)
+IP_WAIT_TIMEOUT=150
+K8S_VM_DEFAULT_MAC=$(virsh domiflist "$K8S_VM_NAME" | grep default | grep -oE '([[:xdigit:]]{2}:){5}[[:xdigit:]]{2}') || true
+if [[ -z $K8S_VM_DEFAULT_MAC ]]; then
+  echo "No MAC address for default interface on VM $K8S_VM_NAME. Check output of \`sudo virsh domiflist $K8S_VM_NAME\`"
+  exit 1
+fi
+while [[ -z $K8S_VM_DEFAULT_IP ]]; do
+  echo "Checking if IP lease exists for interface $K8S_VM_DEFAULT_MAC"
+  K8S_VM_DEFAULT_IP=$(virsh net-dhcp-leases default | grep "$K8S_VM_DEFAULT_MAC" | grep -oE '([0-9]{1,3}\.){3}[0-9]{1,3}') || true
+  if [[ -z $K8S_VM_DEFAULT_IP ]]; then
+    if [[ $IP_WAIT_TIMEOUT -ge 0 ]]; then
+      (( IP_WAIT_TIMEOUT -= 15 ))
+      echo "No IP assigned for $K8S_VM_DEFAULT_MAC. Retrying in 15 seconds"
+      sleep 15
+    else
+      echo "No IP assigned for $K8S_VM_DEFAULT_MAC after 3 minutes. Exiting."
+      exit 1
+    fi
+  fi
+done
+echo "Docker VM IP found: $K8S_VM_DEFAULT_IP. Waiting for sshd to start"
+SSHD_TIMEOUT=60
+while [[ $SSHD_TIMEOUT -ge 0 ]]; do
+  echo '' | nc $K8S_VM_DEFAULT_IP 22 | grep -i ssh && break
+  sleep 5
+  (( SSHD_TIMEOUT -= 5 ))
+done
+ssh-keygen -R $K8S_VM_DEFAULT_IP
+ssh -o StrictHostKeyChecking=no -Ti "$K8S_VM_KEY" vagrant@$K8S_VM_DEFAULT_IP << EOF
+sudo bash -o errexit << EOSU
+hostname $K8S_VM_NAME
+echo $K8S_VM_NAME > /etc/hostname
```

Comment:
Are we assuming SUSE here? If yes, `yast dns edit hostname="${K8S_VM_NAME}"` might work here.

Reply:
This is intended to run on any kvm host which satisfies the requirements. Practically speaking, this will probably be a SLE kvm host every time, but is there a reason to prefer the SUSE-specific hostname rename instead of a more general one?

Reply:
Ah, didn't realize you wanted it that generic. The reason I ask for yast is because it seems to tickle more places and updates the hostname in the prompt faster